box: add runtime lua_call privileges (#10358)
tarantool/tarantool: changelogs/unreleased/gh_10306_lua_call_runtime_access.md

## feature/box

> * Add the ability to grant and revoke access to any `lua_call` function in the
>   internal API (gh-10306).

Reviewer: ```suggestion
* It is now possible to grant and revoke access to any `lua_call` function in the
  internal API (gh-10306).
```
